x-pack/platform/plugins/shared/task_manager/server/task_scheduling.ts

```diff
@@ -284,12 +284,26 @@ export class TaskScheduling {
    */
   public async ensureScheduled(
     taskInstance: TaskInstanceWithId,
-    options?: Record<string, unknown>
+    options?: ScheduleOptions
   ): Promise<TaskInstanceWithId> {
     try {
       return await this.schedule(taskInstance, options);
     } catch (err) {
       if (err.statusCode === VERSION_CONFLICT_STATUS) {
+        // check if task specifies a schedule interval
+        // if so,try to update the just the schedule
+        // only works for interval schedule
+        if (taskInstance.schedule && taskInstance.schedule.interval) {
+          const result = await this.bulkUpdateSchedules([taskInstance.id], taskInstance.schedule);
```

Comment:
`bulkUpdateSchedules` already filters out tasks with no change to the schedule interval so no check needed here.